wordpress-mobile/WordPress-Android
# GB - Add additional check that makes sure the opt-in dialog is shown only once per site

This PR does fix an edge case where the opt-in dialog could be shown more than once on wpcom site (VIPs site with already GB posts on them, but with the set_GB call disabled server side - it's a kind of very edge case, but better to be safe than sorry)

This PR does also:
- Rename `setShowGutenbergInfoPopup ` to `setShowGutenbergInfoPopupForTheNewPosts`, since the old method name was not well chosen
- Remove unused app prefs old methods
- Not quite sure why that `WPScreenshotTest.java` is not failing since those old method is not used anymore in the app and the popup could be shown when opening a GB post

To test:
- Erase your preference on the RC card
- Start the app and make sure the app synch with the backend (tap around, close it, re.open)
- Edit a GB based post. The pop-up should show
- Erase the preference on the RC card (This simulate a setEditor to GB call that fails for a VIP site)
- Restart the app
- Make sure the editor prefs are synched with the server (breakpoint in `refreshCurrentSelectedSiteAfterEditorChanges` [here](https://github.com/wordpress-mobile/WordPress-Android/blob/865d69db1f332a2c3ebdd7a10b689c5040dc3889/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L1179)  and the mobile editor item should be blank
- Re-open a GB based post. 
- Gutenberg is started, but no opt-it dialog is shown on the screen


Update release notes:

- [ x ] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.
